drivers/flash/flash_simulator_stub/Kconfig

```diff
@@ -0,0 +1,25 @@
+# Flash simulator config
+
+# Copyright (c) 2018 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig FLASH_SIMULATOR_STUB
+	bool "Flash simulator stub"
+	depends on !FLASH_SIMULATOR
```

Comment:
should we have this ?
Usually flash drivers are not depending on another driver not being selected.

Of course selecting multiple drivers might end in linking errors later, but if we test for `!FLASH_SIMULATOR` then you should probably also test for `!NORDIC_QSPI_NOR`, `!SPI_NOR` etc.
https://github.com/nrfconnect/sdk-zephyr/blob/a64e96d17cc741937a8f5ff01a824ef0df22f662/drivers/flash/Kconfig.nordic_qspi_nor#L4
https://github.com/nrfconnect/sdk-zephyr/blob/a64e96d17cc741937a8f5ff01a824ef0df22f662/drivers/flash/Kconfig.nor#L5

Reply:
Yes, but other drivers are not cross compatible with other devices. This is alternative driver for the same "zephyr,sim-flash" device.

Reply:
Other driver do not collide with it.

Reply:
It uses same configuration as flash_simulator for make this patch easier to be done.

Reply:
fair enough.